compiler/src/dotty/tools/backend/sjs/JSEncoding.scala

```diff
@@ -207,6 +207,10 @@ object JSEncoding {
        * BoxedUnit$ is a JVM artifact.
        */
       ir.Definitions.BoxedUnitClass
+    } else if(sym == defn.NullClass) {
+      ir.Definitions.NullClass
+    } else if(sym == defn.NothingClass) {
+      ir.Definitions.NothingClass
```

Comment:
That's the spirit of the fix, but it is a bit in the wrong place, because it applies to too many things. In particular, this would apply to `classOf[Null]` or `Array[Null]`, where we want to keep `scala.runtime.Null$`.

The only place where we want to replace `scala.runtime.Null$` by `ir.Definitions.NullClass` is in the signature of methods (and even there, only at the "top level", not in array types). Hence the correct place to do that would be in [`internalName`](https://github.com/lampepfl/dotty/blob/3339991fdf1251d679cd7c069bca23dde00831fb/compiler/src/dotty/tools/backend/sjs/JSEncoding.scala#L413), after getting the result of `toTypeRef` and before passing it to `encodeTypeRef`. That is the correct place to intercept `ClassRef("sr_Null$")` and turn it into `ClassRef(ir.Definitions.NullClass)`.

(of course, ditto for `Nothing`).